weed/storage/volume_vacuum.go

```diff
@@ -463,6 +482,47 @@ func copyDataBasedOnIndexFile(srcDatName, srcIdxName, dstDatName, datIdxName str
 		return err
 	}
 
-	return newNm.SaveToIdx(datIdxName)
+	err = newNm.SaveToIdx(datIdxName)
+	if err != nil {
+		return err
+	}
+	if v.needleMapKind == NeedleMapInMemory {
+		nm := &NeedleMap{
+			m: needle_map.NewCompactMap(),
+		}
+		indexFile, err := os.OpenFile(datIdxName, os.O_RDWR|os.O_CREATE, 0644)
+		if err != nil {
+			glog.Errorf("cannot open Volume Index %s: %v", datIdxName, err)
+			return err
+		}
+		defer indexFile.Close()
 
+		//can be optimized, filling nm in  oldNm.AscendingVisit
+		v.tmpNm, err = doOffsetLoading(indexFile, 0, nm)
```

Comment:
close previous v.tmpNm

Reply:
the same logic can be applied to leveldb indexes. Please consider refactoring.

Reply:
Sure,  I'll refactor these logic and then repen this PR again.
Now I convert this PR to draft.

Reply:
I have refactored and tested on my server  on both memory index and leveldb index.